thrift/thrift.bzl

```diff
@@ -0,0 +1,62 @@
+"""Rules for organizing thrift files."""
+
+load("//scala:scala.bzl", "implicit_deps")
+
+_thrift_filetype = FileType([".thrift"])
+
+#TODO some unanswered questions...
+# - should the tar for a given target include the deps? I believe the answer is no (they should be transitive)
+# - should "building" this tar trigger the builds of the dependent tars? I believe the answer should be yes...not sure how to trigger that?
+#   hmm, on the other side, though... perhaps the answer is no, and it should only be triggered when it is actually used/needed? yes, that
+#   seems more in line with bazel philosophy of "only build what you need when you need it"
+def _thrift_library_impl(ctx):
+  #TODO how do we want to archive this? could just do another jar to be consistent with everything else...
+  cmd = "tar -c $@ > {out}".format(out=ctx.outputs.libarchive.path)
+  ctx.action(
+    inputs = ctx.files.srcs,
+    outputs = [ctx.outputs.libarchive],
+    command = cmd,
+    progress_message = "making thrift archive %s" % ctx.label,
+    arguments = [f.path for f in ctx.files.srcs],
+  )
+  transitive_archives = _collect_thrift_tars(ctx.attr.deps)
+  transitive_archives += [ctx.outputs.libarchive]
+
+  transitive_srcs = _collect_thrift_srcs(ctx.attr.deps)
+  transitive_srcs += ctx.files.srcs
+  return struct(
```

Comment:
can we do the nesting pattern here under `thrift` so we can see the "type" of this thing later?

`struct(thrift = struct(...)``